lib/net/ssh/proxy/command.rb

```diff
@@ -93,7 +93,7 @@ def io.recv(size)
             result = read_nonblock(size)
           rescue IO::WaitReadable, Errno::EINTR
             timeout_in_seconds = 20
-            if IO.select([self], nil, [self], timeout_in_seconds) == nil
+            if wait_readable(timeout_in_seconds) == nil
```

Comment:
No prob, take your time. Just wanted to sort the timeout issue out. One thing I don't understand from the original is, why was the io included in the error array argument, as the code seems to only be interested in whether the socket is readable.